From 40b86566a93ccb4c4af8692a8687b1a4b063bf29 Mon Sep 17 00:00:00 2001
From: Giulio Camuffo <giulio.camuffo@jollamobile.com>
Date: Mon, 7 Jul 2014 17:54:04 +0300
Subject: [PATCH] Add cmake test for QtCompositor and disable cmake files for
 QtWaylandClient

take only the very minimal version of cmake test until we upgrade to 5.4

Change-Id: Id6e2d46b55e521218a3c0f44cf9a4a2be5f0df7f
Reviewed-by: Robin Burchell <robin+qt@viroteck.net>

Conflicts:
	src/client/client.pro
	src/compositor/compositor_api/qwaylandbufferref.h
	src/compositor/compositor_api/qwaylandquicksurface.h
	src/compositor/compositor_api/qwaylandsurface.cpp
	src/compositor/wayland_wrapper/qwlcompositor.cpp
	src/compositor/wayland_wrapper/qwlsurface.cpp
	tests/auto/auto.pro

Conflicts:
	tests/auto/auto.pro
---
 src/client/client.pro           |  2 ++
 tests/auto/auto.pro             |  1 +
 tests/auto/cmake/CMakeLists.txt | 15 +++++++++++++++
 tests/auto/cmake/cmake.pro      |  6 ++++++
 4 files changed, 24 insertions(+)
 create mode 100644 tests/auto/cmake/CMakeLists.txt
 create mode 100644 tests/auto/cmake/cmake.pro

diff --git a/src/client/client.pro b/src/client/client.pro
index 10cbd31..18fcd23 100644
--- a/src/client/client.pro
+++ b/src/client/client.pro
@@ -7,6 +7,8 @@ MODULE_PLUGIN_TYPES = wayland-graphics-integration-client
 
 load(qt_module)
 
+CONFIG -= precompile_header
+CONFIG -= create_cmake
 CONFIG += link_pkgconfig qpa/genericunixfontdatabase wayland-scanner
 
 !equals(QT_WAYLAND_GL_CONFIG, nogl) {
diff --git a/tests/auto/auto.pro b/tests/auto/auto.pro
index 1d7d930..822fdeb 100644
--- a/tests/auto/auto.pro
+++ b/tests/auto/auto.pro
@@ -5,4 +5,5 @@ TEMPLATE=subdirs
 contains(CONFIG, wayland-compositor) {
     SUBDIRS += compositor
     SUBDIRS += client
+    SUBDIRS += cmake
 }
diff --git a/tests/auto/cmake/CMakeLists.txt b/tests/auto/cmake/CMakeLists.txt
new file mode 100644
index 0000000..13d9031
--- /dev/null
+++ b/tests/auto/cmake/CMakeLists.txt
@@ -0,0 +1,15 @@
+
+cmake_minimum_required(VERSION 2.8)
+
+project(qmake_cmake_files)
+
+enable_testing()
+
+find_package(Qt5Core REQUIRED)
+set(Qt5_MODULE_TEST_DEPENDS Quick)
+
+include("${_Qt5CTestMacros}")
+
+#test_module_includes(
+#  Compositor QWaylandBufferRef
+#)
diff --git a/tests/auto/cmake/cmake.pro b/tests/auto/cmake/cmake.pro
new file mode 100644
index 0000000..5098c2c
--- /dev/null
+++ b/tests/auto/cmake/cmake.pro
@@ -0,0 +1,6 @@
+# Cause make to do nothing.
+TEMPLATE = subdirs
+
+CMAKE_QT_MODULES_UNDER_TEST = waylandclient
+
+CONFIG += ctest_testcase
-- 
2.1.1

